Userland/Libraries/LibCrypto/Curves/SECP256r1.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, Michiel Visser <opensource@webmichiel.nl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <AK/UFixedBigInt.h>
+
+namespace Crypto::Curves {
+
+struct JacobianPoint {
+    u256 x { 0u };
+    u256 y { 0u };
+    u256 z { 0u };
+};
+
+class SECP256r1 {
+public:
+    static ErrorOr<ByteBuffer> generate_public_key(ReadonlyBytes a);
+    static ErrorOr<ByteBuffer> compute_coordinate(ReadonlyBytes scalar_bytes, ReadonlyBytes point_bytes);
+
+private:
+    static u256 modular_reduce(u256 const& value);
```

Comment:
Is having a class for each curve how it's done for the rest of these curves? Nothing about this class requires it to actually be a class. This whole header could be changed to be 
```cpp
#include <AK/ByteBuffer.h>
#include <AK/ErrorOr.h>
namespace Crypto::Curves {
namespace SECP256r1 }
 ErrorOr<ByteBuffer> generate_public_key(ReadonlyBytes a);
 ErrorOr<ByteBuffer> compute_coordinate(ReadonlyBytes scalar_bytes, ReadonlyBytes point_bytes);
}
}
```
with everything else (including struct JacobianPoint) moved into the .cpp file.

Reply:
Right now, the only other implementation is the X25519 curve, which has the same style with a class and a bunch of private methods.
I'm not that used to programming in C++, so I don't know what the preferred way of doing this is.
I agree that this is possible too, and might be nicer and simpler.

Reply:
Let's keep it the class like this for right now, as I have another pull request in the pipeline where I make the elliptic curve implementations extend a base class to simplify the implementation in LibTLS.
I agree that the private methods and constants can moved into the cpp file, reducing the mess in the header.
I will address this in the next PR.